### PR TITLE
Fix attributes of optimized model

### PR DESCRIPTION
**Pull request by AlexanderDokuchaev**
- Set `ModelOptimizationType.NNCF` instead `ModelOptimizationType.MO`  for exported model that optimized by NNCF
- Use `ModelOptimizationType.POT` instead `OptimizationType.POT`

CI: 658 🟢

### CODE REVIEW

File: mmdet/apis/ote/apis/detection/nncf_task.py [hunk 1]
```diff
@@ -65,6 +66,7 @@ def __init__(self, task_environment: TaskEnvironment):
         self._val_dataloader = None
         self._compression_ctrl = None
         self._nncf_preset = "nncf_quantization"
+        self._optimization_type = ModelOptimizationType.NNCF
```
Comment (druzhkov-paul) [inline mmdet/apis/ote/apis/detection/nncf_task.py]: What should be the optimization type for the model, which is exported out of the NNCF task?

File: mmdet/apis/ote/apis/detection/nncf_task.py [hunk 2]
```diff
@@ -65,6 +66,7 @@ def __init__(self, task_environment: TaskEnvironment):
...
         check_nncf_is_enabled()
         super().__init__(task_environment)
```
Comment (druzhkov-paul) [inline mmdet/apis/ote/apis/detection/nncf_task.py]: Is it Ok to call parent class constructor after setting custom optimization type here?

File: mmdet/apis/ote/apis/detection/nncf_task.py (hunk 1 above)
Comment (AlexanderDokuchaev) [reply mmdet/apis/ote/apis/detection/nncf_task.py]: After optimize and export optimization_type should be NNCF

File: mmdet/apis/ote/apis/detection/nncf_task.py (hunk 2 above)
Comment (AlexanderDokuchaev) [reply mmdet/apis/ote/apis/detection/nncf_task.py]: ofc no, thanks

File: mmdet/apis/ote/apis/detection/nncf_task.py (hunk 1 above)
Comment (druzhkov-paul) [reply mmdet/apis/ote/apis/detection/nncf_task.py]: Hm...

File: mmdet/apis/ote/apis/detection/nncf_task.py [hunk 3]
```diff
@@ -119,10 +121,6 @@ def _load_model(self, model: ModelEntity):
 
             self.confidence_threshold = model_data.get('confidence_threshold',
                 self._hyperparams.postprocessing.confidence_threshold)
-            if model_data.get('anchors'):
-                anchors = model_data['anchors']
-                self._config.model.bbox_head.anchor_generator.heights = anchors['heights']
-                self._config.model.bbox_head.anchor_generator.widths = anchors['widths']
```
Comment (druzhkov-paul) [inline mmdet/apis/ote/apis/detection/nncf_task.py]: I'd not remove it. This might be the reason of the test failure.

**Review by druzhkov-paul — APPROVED** [review]

**Comment by druzhkov-paul** [comment]
Build 658 ✔️ 
Build 659 🔴

**Comment by druzhkov-paul** [comment]
@AlexanderDokuchaev please fix tests.

**Review by LeonidBeynenson — APPROVED** [review]

**Comment by druzhkov-paul** [comment]
Build 661 ✔️

**Pull request merged**